src/popup/router/components/InputField.vue

```diff
@@ -1,35 +1,96 @@
 <template>
-  <InputWrapper
+  <div
     class="input-field"
-    :class="{ readonly }"
-    v-bind="$attrs"
+    :class="{ 'new-ui': newUi }"
   >
-    <slot
-      v-for="slot in Object.keys($slots)"
-      :slot="slot"
-      :name="slot"
-    />
-    <input
-      class="input"
-      v-bind="$attrs"
-      autocomplete="off"
-      :value="value"
-      :data-cy="$attrs.type ? `input-${$attrs.type}` : 'input'"
-      :disabled="readonly"
-      step="any"
-      @input="$emit('input', $event.target.value)"
+    <label
+      v-if="label || $slots.label"
+      :for="inputId"
+      class="label"
     >
-  </InputWrapper>
+      <slot name="label">{{ label }}</slot>
+    </label>
+
+    <label
+      data-cy="input-wrapper"
+      class="input-wrapper"
+      :class="{ error, warning, plain, readonly }"
+    >
+      <div class="main-inner">
+        <slot
+          v-if="!error && !warning"
+          name="left"
+        />
+        <StatusIcon
+          v-else
+          :status="error && 'alert' || warning && 'warning'"
+        />
+        <slot
+          :id="_uid"
+          :input-id="inputId"
+        >
+          <input
+            :id="inputId"
+            v-bind="$attrs"
+            class="input"
+            autocomplete="off"
+            step="any"
+            :value="value"
+            :class="{ 'new-ui': newUi }"
+            :data-cy="$attrs.type ? `input-${$attrs.type}` : 'input'"
+            :disabled="readonly"
+            @input="$emit('input', $event.target.value)"
+            @focusin="focused = true"
+            @focusout="focused = false"
+          >
+        </slot>
+        <slot name="right" />
+      </div>
+
+      <div class="under">
+        <slot
+          name="under"
+          :focused="focused"
+        />
+      </div>
+    </label>
+
+    <label
+      v-if="error && errorMessage || warning && warningMessage"
+      class="message"
+      :class="{ error, warning }"
+      :for="inputId"
+    >
+      {{ error ? errorMessage : warningMessage }}
+    </label>
+  </div>
 </template>
 
 <script>
-import InputWrapper from './InputWrapper.vue';
+import StatusIcon from './StatusIcon.vue';
 
 export default {
-  components: { InputWrapper },
+  components: {
+    StatusIcon,
+  },
   props: {
     value: { type: [String, Number], default: null },
+    label: { type: String, default: '' },
+    error: Boolean,
+    errorMessage: { type: String, default: '' },
+    warning: Boolean,
+    warningMessage: { type: String, default: '' },
     readonly: Boolean,
+    plain: Boolean,
+    newUi: Boolean,
```

Comment:
I'd suggest to group the props by type

Reply:
I kind of agree, but now the props are grouped by the purpose. error + errorMessage.